src/log/ModDaemon.cc

```diff
@@ -298,29 +298,34 @@ logfile_mod_daemon_writeline(Logfile * lf, const char *buf, size_t len)
         }
         return;
     }
-    /* Append this data to the end buffer; create a new one if needed */
+
     /* Are we eol? If so, prefix with our logfile command byte */
+    if (ll->eol) {
+	char tb[2];
+	tb[0] = 'L';
+	tb[1] = '\0';
+	logfile_mod_daemon_append(lf, tb, 1);
+	ll->eol = 0;
+    }
+
+    /* Append this data to the end buffer; create a new one if needed */
     logfile_mod_daemon_append(lf, buf, len);
 }
 
 static void
 logfile_mod_daemon_linestart(Logfile * lf)
 {
-    l_daemon_t *ll = static_cast<l_daemon_t *>(lf->data);
-    char tb[2];
-    assert(ll->eol == 1);
-    ll->eol = 0;
-    tb[0] = 'L';
-    tb[1] = '\0';
-    logfile_mod_daemon_append(lf, tb, 1);
+    /* 
+     * Sending the starting command byte has been moved to
+     * logfile_mod_daemon_writeline
+     */
 }
 
 static void
 logfile_mod_daemon_lineend(Logfile * lf)
 {
     l_daemon_t *ll = static_cast<l_daemon_t *>(lf->data);
     logfile_buffer_t *b;
-    assert(ll->eol == 0);
     ll->eol = 1;
```

Comment:
Would it be possible to do nothing if `ll->eol` is still true? If yes, please do that instead of just removing the no-longer-valid assertion. Here is a sketch:

```C++
logfile_buffer_t *b;
if (ll->eol == 1) // logfile_mod_daemon_writeline() wrote nothing
    return;
```

Technically, the C-style `b` declaration should be moved into the `if` statement instead of being left where it is, but that polishing change would obscure the PR diff too much IMO. Thus, I do not recommend moving it. I wrote it above to show the placement of the new `if` statement (replacing the old assertion).

Reply:
Done